### DISCUSSION TITLE
Optionally disable `inspect_site` to save credits

### DISCUSSION THREAD

**Discussion by sabslikesobs** [cline/cline#452] [Feature Requests]
`inspect_site` really makes my credit usage blow up -- I made tons of changes on about $5, then started using `inspect_site` and did almost nothing to blow through another $5. (I wonder if the prompt cache gets mixed up or something). 

I don't want Claude-Dev to open `inspect_site`, but I do want normal files to be read automatically, because they don't cost much. Can you add a separate toggle in the options to disable or force-ask-me-every-time for `inspect_site`?

I've worked around this in the meantime by breaking my devcontainer so that `inspect_site` doesn't work.

**Comment by TechMitten** [comment]
> I've worked around this in the meantime by breaking my devcontainer so that `inspect_site` doesn't work.
>
@sabslikesobs Can you please explain this, I'm interested.

**Reply by sabslikesobs** [reply]
My devcontainer from Ruby on Rails' `rails new` didn't support `inspect_site` in the first place. I fixed it, found out about the credits, and then reverted my fix. 

`inspect_site` depends on puppeteer [and its dependencies](https://ploi.io/documentation/server/how-to-install-puppeteer-on-ubuntu), so I installed those to get `inspect_site` working in the first place, then removed them once I found out I didn't want it.

I'm not sure how to intentionally break `inspect_site` if the system already has those dependencies installed.
